### ISSUE TITLE
Unable to load distributed optimizer in TF2

### ISSUE THREAD

**Issue by gchesnokov** [horovod/horovod#2486]
**Environment:**
1. Framework: TensorFlow, Keras
2. Framework version: tensorflow==1.12.3, tensorflow==2.1.1
3. Horovod version: horovod==0.18.1
7. Python version: 2.7.17

**Bug report:**
I've stumbled upon a problem with loading a distributed optimizer in Tensorflow.keras. Workflow used to work with tf1.12, but seems to be broken in tf2.1.
Minimal example to reproduce:
```
import horovod.tensorflow.keras as hvd
import numpy as np
import tensorflow as tf
from tensorflow.python.keras import backend as K


def create_model():
    opt = tf.keras.optimizers.SGD(lr=0.01 * hvd.size(), momentum=0.9)
    opt = hvd.DistributedOptimizer(opt)

    model = tf.keras.models.Sequential()
    model.add(tf.keras.layers.Dense(2, input_shape=(3,)))
    model.add(tf.keras.layers.RepeatVector(3))
    model.add(tf.keras.layers.TimeDistributed(tf.keras.layers.Dense(3)))
    model.compile(
        loss=tf.keras.losses.MSE,
        optimizer=opt,
        metrics=[tf.keras.metrics.categorical_accuracy],
        sample_weight_mode='temporal',
        experimental_run_tf_function=False
    )
    return model


def generator():
    while 1:
        yield (np.random.random((1, 3)), np.random.random((1, 3, 3)))


def train(filename):
    model = create_model()
    callbacks = [hvd.callbacks.BroadcastGlobalVariablesCallback(0)]
    model.fit(generator(), steps_per_epoch=10, callbacks=callbacks, epochs=1)
    if hvd.rank() == 0:
        model.save(filename)


def load(filename):
    if hvd.rank() == 0:
        model = hvd.load_model(filename)
    else:
        model = create_model()
    print('rank {}, opt {}'.format(hvd.rank(), model.optimizer))


hvd.init()
filename = 'test_model.h5'
train(filename)
load(filename)
```
While using tf1.12 I get wrapped optimizers after loading:
```
[1,1]<stdout>:rank 1, opt <horovod._keras.SGD object at 0x7f2b8f735750>
[1,0]<stdout>:rank 0, opt <horovod._keras.SGD object at 0x7f93b45a2750>
```
With tf2.1:
```
[1,1]<stdout>:rank 1, opt <horovod._keras.SGD object at 0x7efd0623c110>
[1,0]<stdout>:rank 0, opt <tensorflow.python.keras.optimizer_v2.gradient_descent.SGD object at 0x7f30a8016d90>
```
I've tried to modify load like this: `model = hvd.load_model(filename, custom_optimizers=[tf.keras.optimizers.SGD])`
But got another error:
```
[1,0]<stderr>:Traceback (most recent call last):
[1,0]<stderr>:  File "test_horovod.py", line 49, in <module>
[1,0]<stderr>:    load(filename)
[1,0]<stderr>:  File "test_horovod.py", line 40, in load
[1,0]<stderr>:    model = hvd.load_model(filename, custom_optimizers=[tf.keras.optimizers.SGD])
[1,0]<stderr>:  File "/usr/lib/python2.7/dist-packages/horovod/tensorflow/keras/__init__.py", line 156, in load_model
[1,0]<stderr>:    return _impl.load_model(keras, wrap_optimizer, filepath, custom_optimizers, custom_objects)
[1,0]<stderr>:  File "/usr/lib/python2.7/dist-packages/horovod/_keras/__init__.py", line 123, in load_model
[1,0]<stderr>:    return keras.models.load_model(filepath, custom_objects=horovod_objects)
[1,0]<stderr>:  File "/usr/lib/python2.7/dist-packages/tensorflow_core/python/keras/saving/save.py", line 146, in load_model
[1,0]<stderr>:    return hdf5_format.load_model_from_hdf5(filepath, custom_objects, compile)
[1,0]<stderr>:  File "/usr/lib/python2.7/dist-packages/tensorflow_core/python/keras/saving/hdf5_format.py", line 193, in load_model_from_hdf5
[1,0]<stderr>:    model._make_train_function()
[1,0]<stderr>:  File "/usr/lib/python2.7/dist-packages/tensorflow_core/python/keras/engine/training.py", line 2057, in _make_train_function
[1,0]<stderr>:    params=self._collected_trainable_weights, loss=self.total_loss)
[1,0]<stderr>:  File "/usr/lib/python2.7/dist-packages/tensorflow_core/python/keras/optimizer_v2/optimizer_v2.py", line 503, in get_updates
[1,0]<stderr>:    grads = self.get_gradients(loss, params)
[1,0]<stderr>:  File "/usr/lib/python2.7/dist-packages/tensorflow_core/python/keras/optimizer_v2/optimizer_v2.py", line 389, in get_gradients
[1,0]<stderr>:    "/gradients"):
[1,0]<stderr>:TypeError: unsupported operand type(s) for +: 'function' and 'str'
```

**Comment by tgaddair** [comment]
Hey @gchesnokov, can you try upgrading to a more recent version of Horovod?  I am unable to repro the issue using the latest Horovod version (0.21.0).

**Comment by gchesnokov** [comment]
Thanks for answer, @tgaddair. After a deeper dive I've discovered the cause of my issue. Turns out it was fixed in #1935, so I'll consider upgrading Horovod version. Thanks again
